src/components/Popup/Popup.tsx

```diff
@@ -80,78 +85,61 @@ export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupSt
 
     /** Element to be rendered in-place where the popup is defined. */
     trigger: PropTypes.node,
-
-    /** Additional CSS styles to apply to the component instance.  */
-    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-
-    /** Override for theme site variables to allow modifications of component styling via themes. */
-    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-
-    /** Accessibility behavior if overridden by the user. */
-    accessibility: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
   }
 
   public static defaultProps: IPopupProps = {
-    as: Portal,
+    accessibility: PopupBehavior,
     align: 'start',
     position: 'above',
-    accessibility: PopupBehavior,
   }
 
-  public state = { triggerRef: undefined, popupOpened: false }
+  public static autoControlledProps = ['open']
 
   protected actionHandlers: AccessibilityActionHandlers = {
-    performClick: e => this.onTriggerClick(e),
-    closeAndFocusTrigger: e => this.handlePopupState(e, () => false, this.focusTrigger),
+    toggle: e => _.invoke(this.props, 'onOpenChange', e, !this.props.open),
+    closeAndFocusTrigger: e => {
+      _.invoke(this.props, 'onOpenChange', false)
+      _.invoke(this.state.triggerRef, 'focus')
```

Comment:
So, in case of user won't provide ```onOpenChange``` function which will handle open/close state, the Popup won't be closed, but the trigger would be focused. It seems like a not expected scenario.
And seems like same for toggle (in case it's not a button).
Let's discuss it.

Reply:
Speaking of a toggle, if a trigger is not ```button``` element,  you won't be able to toggle popup with ```Enter```\ ```Space``` button. So, do we expect here that a user should use ```onOpenChange`` function in case to handle toggling?

Reply:
yes - but this leads us to more general problem that is rather accessibility related. In this PR I would like to focus on the following things, **with all the previously introduced functionality being untouched**:
- refactor and simplify `Popup` base implementation
- introduce controlled mode to it

So, these are the main reasons I've left `PopupBehavior`'s functionality as it was introduced before. so that If we would like to address/discuss some problems related to this aspect exclusively, we would be able to do it by means of a dedicated PR.

Please, let me know about your thoughts on that :)